deps/build.jl

```diff
@@ -1,65 +1,79 @@
-
-    LDFLAGS = "-Wl,-rpath,$vdir/lib -Wl,-R$vdir/lib -Wl,-R$nemovdir/lib -Wl,-R\$\$ORIGIN/../share/julia/site/v$(VERSION.major).$(VERSION.minor)/Polymake/local/lib"
-
-    cd(wdir)
-
-    const polymake = joinpath(wdir, "polymake")
-
-    try
-      run(`git clone https://github.com/polymake/polymake.git`)
-    catch
-      cd(polymake)
-      try
-         run(`git pull --rebase`)
-      catch
-      end
-      cd(wdir)
+# test whether polymake config is available in path
+pm_config = nothing
+try
+    global pm_config = chomp(read(`which polymake-config`, String))
+catch
+    if haskey(ENV, "POLYMAKE_CONFIG")
+        global pm_config = ENV["POLYMAKE_CONFIG"]
     end
-
-    cd(polymake)
-
-    withenv("CPP_FLAGS"=>"-I$vdir/include", "LD_LIBRARY_PATH"=>"$vdir/lib:$nemodir/lib") do
-       run(`$polymake/configure --prefix=$vdir --with-gmp=$nemovdir --with-mpfr=$nemovdir`)
-       withenv("LDFLAGS"=>LDFLAGS) do
-          run(`make -j4`)
-          run(`make install`)
-       end
-    end
-
-    ENV["POLYMAKE_CONFIG"] = "$pkgdir/local/bin/polymake-config"
 end
 
-pm_includes = chomp(readstring(`$(ENV["POLYMAKE_CONFIG"]) --includes`))
-pm_includes = map(i->i[3:end], map(String,split(pm_includes)))
-push!(pm_includes,Pkg.dir(pm_includes[1],"..","share","polymake"))
-pm_includes=join(pm_includes," ")
-
-pm_cflags = chomp(readstring(`$(ENV["POLYMAKE_CONFIG"]) --cflags`))
-pm_ldflags = chomp(readstring(`$(ENV["POLYMAKE_CONFIG"]) --ldflags`))
-pm_libraries = chomp(readstring(`$(ENV["POLYMAKE_CONFIG"]) --libs`))
+if pm_config === nothing
+    # TODO: Install polymake, for now just throw an error
+    error("Set `POLYMAKE_CONFIG` ENV variable. And rebuild PolymakeWrap by calling `import Pkg; Pkg.build(\"PolymakeWrap\")`.")
+
+    # This is the old Julia 0.6 script and doesnt work anymore.
+    #
+    # build polymake, using Nemo's GMP and MPFR
+    # const oldwdir = pwd()
+    # const pkgdir = Pkg.dir("PolymakeWrap")
+    # const nemodir = Pkg.dir("Nemo")
+    #
+    # wdir = "$pkgdir/deps"
+    # vdir = "$pkgdir/local"
+    # nemovdir = "$nemodir/local"
+    #
+    # LDFLAGS = "-Wl,-rpath,$vdir/lib -Wl,-R$vdir/lib -Wl,-R$nemovdir/lib -Wl,-R\$\$ORIGIN/../share/julia/site/v$(VERSION.major).$(VERSION.minor)/Polymake/local/lib"
+    #
+    # cd(wdir)
+    #
+    # const polymake = joinpath(wdir, "polymake")
+    #
+    # try
+    #   run(`git clone https://github.com/polymake/polymake.git`)
+    # catch
+    #   cd(polymake)
+    #   try
+    #      run(`git pull --rebase`)
+    #   catch
+    #   end
+    #   cd(wdir)
+    # end
+    #
+    # cd(polymake)
+    #
+    # withenv("CPP_FLAGS"=>"-I$vdir/include", "LD_LIBRARY_PATH"=>"$vdir/lib:$nemodir/lib") do
+    #    run(`$polymake/configure --prefix=$vdir --with-gmp=$nemovdir --with-mpfr=$nemovdir`)
+    #    withenv("LDFLAGS"=>LDFLAGS) do
+    #       run(`make -j4`)
+    #       run(`make install`)
+    #    end
+    # end
+    #
+    # ENV["POLYMAKE_CONFIG"] = "$pkgdir/local/bin/polymake-config"
+end
```

Comment:
Shouldn't we repair the installation?

Reply:
Yes, but I wasn't sure about the right strategy. Depending on Nemo just for GMP and MPFR also somehow seems flawed. Shouldn't we link against the julia GMP and MPFR?

Reply:
linking against Nemo gmp doesn't work, Nemo doesn't ship gmpxx stuff...